workspaces/meewee-time-registration/plugins/plugin-meewee-time-registration/meewee-time-registration/src/request/api/azdevops/AzDevOpsClient.ts

```diff
@@ -0,0 +1,131 @@
+import { DiscoveryApi, FetchApi, OAuthApi } from '@backstage/core-plugin-api';
+import { AzDevopsApi, request } from '../../../types/azdevops/types';
+import { ResponseError } from '@backstage/errors';
+
+/** @public */
+export class AzDevOpsClient implements AzDevopsApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly authApi: OAuthApi;
+  private readonly client: FetchApi;
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    authApi: OAuthApi;
+    fetchApi: FetchApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.authApi = options.authApi;
+    this.client = options.fetchApi;
+  }
+
+  private async handleResponse(req: request): Promise<any> {
+    // azure devops scope  '499b84ac-1321-427f-aa17-267ca6975798/.default'
+    const token = await this.authApi.getAccessToken(req.scope);
+
+    const temp: any = {};
+
+    if (token) {
+      temp.Authorization = `Bearer ${token}`;
+    }
+
+    const response = await this.client.fetch(req.url.toString(), {
+      method: req.method,
+      headers: temp,
+    });
+
+    if (!response.ok) {
+      throw await ResponseError.fromResponse(response);
+    }
+
+    return await response.json();
+  }
+
+  async getProjects(): Promise<any> {
+    const url = `${await this.discoveryApi.getBaseUrl('ado')}/projects`;
```

Comment:
What is 'ado'? This is expecting a plugin Id so not sure where this is defined, haven't seen it in this PR?

Reply:
Same with other cases of this in this file

Reply:
ado is for azure devops feature that we're planning to integrate with the MeeWee plugin. This is a feature that we're planning to implement in the future. Sorry we forgot to remove the codes related for that feature.